src/Database/Statement/BufferedStatement.php

```diff
@@ -106,13 +106,21 @@ public function fetch($type = 'num')
         return $this->_records[] = $record;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function fetchAssoc()
```

Comment:
Do we need this method if we can simply do `fetch(..::class)`?

Reply:
The only thing better than method('string') would be  method(static::SOME_CONST) and better still would be methodConst(). It's just easier to call. 

My guess is that the original issue (11795) was about wanting more access to PDO statement object. Dbal and other lightweight PDO abstractions generally provide these short cut methods fetchAssoc/fetchColumn/fetchObject.

Reply:
I can see your point. 

Doesnt fetch() = fetchNum() because num is the default.

Reply:
Sure, there is a default on fetch(), but do use cases warrant an own method for a non default option that can easily be done using the primary one?

Reply:
Doesn't seem to be a right or wrong answer here for me, It seems a matter of preference. I get what you are saying. I like the methods they are easy to find in in my ide and I don't have to know about the constants or strings. These new methods are about convenience to me. 

Someone let me know and I throw it one way or the other.

Reply:
I let @markstory or @lorenzo decide on it as well.

Reply:
I think having `fetchAssoc()` is reasonable as the use cases of `fetchNum()` are low enough that it doesn't warrant having a method right now. `fetchObj()` was removed because it causes problems with `Collection` related methods.

Reply:
Leaving the new methods/constants as they are in this commit.